### PR TITLE
[FIXED JENKINS-42952] Make currentBuild.duration meaningful

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-42952](https://issues.jenkins-ci.org/browse/JENKINS-42952)

...and add currentBuild.durationString.

cc @reviewbybees esp @jglick

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java [hunk 1]
```diff
@@ -133,7 +133,12 @@ public long getStartTimeInMillis() throws AbortException {
 
     @Whitelisted
     public long getDuration() throws AbortException {
-	 return build().getDuration();
+	 return System.currentTimeMillis() - build().getStartTimeInMillis();
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java]: This will be wrong for a build other than `currentBuild`.

Comment (abayer) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java]: Updated.

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapperTest.java [hunk 2]
```diff
@@ -166,6 +167,21 @@
...
+    @Test public void duration() {
+        r.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = r.j.createProject(WorkflowJob.class, "this-job");
+                p.setDefinition(new CpsFlowDefinition(
+                        "echo \"currentBuild.duration='${currentBuild.duration}'\"\n" +
+                                "echo \"currentBuild.durationString='${currentBuild.durationString}'\"\n", true));
+                WorkflowRun b = r.j.assertBuildStatusSuccess(p.scheduleBuild2(0).get());
+                r.j.assertLogNotContains("currentBuild.duration='0'", b);
+                r.j.assertLogNotContains("currentBuild.durationString='" + Messages.Run_NotStartedYet() + "'", b);
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapperTest.java]: Should this not be `Run_InProgressDuration`?

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapperTest.java [hunk 3]
```diff
@@ -166,6 +167,21 @@
...
 
+    @Issue("JENKINS-42952")
+    @Test public void duration() {
+        r.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = r.j.createProject(WorkflowJob.class, "this-job");
+                p.setDefinition(new CpsFlowDefinition(
+                        "echo \"currentBuild.duration='${currentBuild.duration}'\"\n" +
+                                "echo \"currentBuild.durationString='${currentBuild.durationString}'\"\n", true));
+                WorkflowRun b = r.j.assertBuildStatusSuccess(p.scheduleBuild2(0).get());
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapperTest.java]: `r.j.buildAndAssertSuccess(p)`

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapperTest.java [hunk 4]
```diff
@@ -166,6 +167,21 @@
...
+                r.j.assertLogNotContains("currentBuild.duration='0'", b);
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapperTest.java]: What if it in fact took a few milliseconds to get that far? Would the test not fail?

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapperTest.java (hunk 2 above)
Comment (abayer) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapperTest.java]: re: this and the above - I think you're reading it wrong - it's `assertLogNotContains`, not `assertLogContains`.

Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapperTest.java]: Ah.

**Pull request merged**